src/devices/cpu/mcs51/mcs51.cpp

```diff
@@ -2059,8 +2059,13 @@ void mcs51_cpu_device::execute_run()
 
 		/* decrement the timed access window */
 		if (m_features & FEATURE_DS5002FP)
+		{
 			m_ds5002fp.ta_window = (m_ds5002fp.ta_window ? (m_ds5002fp.ta_window - 1) : 0x00);
 
+			if (m_ds5002fp.rnr_delay > 0)
+				m_ds5002fp.rnr_delay--;
```

Comment:
I can understand that this random number generator is busy for a while, but it does not make sense that it has 1 delay per opcode. I don't know the opcode timings, but hypothetically let's say you have 160 opcodes each 10 cycles, meaning the random number generator is ready after 1600 cycles. Or you have 160 opcodes each 30 cycles, and it's ready after 4800 cycles?

In short, subtract m_inst_cycles for this delay thing?

Reply:
made the changes for the part you asked.  honestly don't know about ta.window, not my code.

Reply:
Good, did you check that it still works as expected or does the delay need to be increased?

Reply:
it appears to be fine